bats/tests/profile/create-profile-output.bats

```diff
@@ -0,0 +1,284 @@
+    assert_success
+    assert_output --partial '[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\locked\application]'
+}
+
+@test 'generates registry output for hkcu/defaults' {
+    run rdctl create-profile --output reg --hive=Hkcu --from-settings
+    assert_success
+    assert_output --partial '[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\application]'
+    run rdctl create-profile --output reg --hive=hkcu --type=Defaults --from-settings
+    assert_success
+    assert_output --partial '[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\application]'
+}
+
+@test 'generates registry output for hkcu/locked' {
+    run rdctl create-profile --output reg --hive=HKCU --type=locked --from-settings
+    assert_success
+    assert_output --partial '[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\locked\application]'
+}
+
+# The result of the `assert_output` for here-documents looks suspicious (I see it always passing),
+# but this serves to document the expected full reg output
+@test 'generates registry output from settings' {
+    run rdctl create-profile --output reg --from-settings
+    assert_success
+    # Just match a few of the lines near the start and the end of the output.
+    # The unit tests do more comprehensive output checking.
+    assert_output --partial '[HKEY_LOCAL_MACHINE\SOFTWARE\Policies]'
+    assert_output --partial '"adminAccess"=dword:0'
+    assert_output --partial '[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\defaults\diagnostics]'
+    assert_output --partial '"showMuted"=dword:0'
+}
+
+@test 'complains when no output type is specified' {
+    run rdctl create-profile --from-settings
+    assert_failure
+    assert_output --partial $"an '--output FORMAT' option of either \"plist\" or \"reg\" must be specified"
+}
+
+@test 'complains when an invalid output type is specified' {
+    run rdctl create-profile --from-settings --output=cabbage
+    assert_failure
+    assert_output --partial $"received unrecognized '--output FORMAT' option of cabbage; \"plist\" or \"reg\" must be specified"
+}
+
+@test 'generates plist output from settings' {
+    run rdctl create-profile --output plist --from-settings
+    assert_success
+    # Just match a few of the lines near the start and the end of the output.
+    # The unit tests do more comprehensive output checking, as do the
+    # tests that work on a subset of the JSON input.
+    assert_output --partial '<?xml version="1.0" encoding="UTF-8"?>'
+    assert_output --partial '<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">'
+    assert_output --partial '<plist version="1.0">'
+    assert_output --partial '    <key>application</key>'
+    assert_output --partial '</plist>'
+}
+
+@test 'verify plutil is ok with the generated plist output' {
+    if ! is_macos; then
+        skip
+    fi
+    run bash -o pipefail -c "rdctl create-profile --output plist --from-settings | plutil -s -"
```

Comment:
I would try to avoid pipelines as much as possible in tests, just because it is hard to see which part of the pipeline fails. So I would prefer the same logic you use in a later test:

```suggestion
    run rdctl create-profile --output plist --from-settings
    assert_success
    plutil -s <<<"$output"
```